src/volta-shim.rs

```diff
@@ -1,35 +1,9 @@
-mod common;
+use std::process::{Command, ExitCode};
 
-use common::{ensure_layout, Error, IntoResult};
-use volta_core::error::{report_error, ExitCode};
-use volta_core::log::{LogContext, LogVerbosity, Logger};
-use volta_core::run::execute_shim;
-use volta_core::session::{ActivityKind, Session};
-use volta_core::signal::setup_signal_handler;
-
-pub fn main() {
-    Logger::init(LogContext::Shim, LogVerbosity::Default)
-        .expect("Only a single Logger should be initialized");
-    setup_signal_handler();
-
-    let mut session = Session::init();
-    session.add_event_start(ActivityKind::Tool);
-
-    let result = ensure_layout().and_then(|()| execute_shim(&mut session).into_result());
+pub fn main() -> ExitCode {
+    let result = Command::new("volta-runner").args(std::env::args()).status();
     match result {
-        Ok(()) => {
-            session.add_event_end(ActivityKind::Tool, ExitCode::Success);
-            session.exit(ExitCode::Success);
-        }
-        Err(Error::Tool(code)) => {
-            session.add_event_tool_end(ActivityKind::Tool, code);
-            session.exit_tool(code);
-        }
-        Err(Error::Volta(err)) => {
-            report_error(env!("CARGO_PKG_VERSION"), &err);
-            session.add_event_error(ActivityKind::Tool, &err);
-            session.add_event_end(ActivityKind::Tool, err.exit_code());
-            session.exit(ExitCode::ExecutionFailure);
-        }
+        Ok(_) => ExitCode::SUCCESS,
+        Err(_) => ExitCode::FAILURE,
```

Comment:
I think this logic may be swallowing the actual exit code: If the `status` call returns `Ok`, that doesn't necessarily mean the command returned success, rather that it was successfully executed.

I believe in the `Ok` case, we probably need to look at the [`ExitStatus::code`](https://doc.rust-lang.org/std/process/struct.ExitStatus.html#method.code) method and forward along the underlying exit code.